zip-0213.rst

```diff
@@ -0,0 +1,179 @@
+order to enable miners to start using Sapling addresses.
+
+
+Specification
+=============
+
+Prior to activation of the TODO network upgrade, the existing consensus rule requiring
+coinbase transactions to have zero Sapling outputs is enforced.
+
+Once the TODO network upgrade activates:
+
+- The above existing consensus rule is no longer active, and coinbase transactions MAY
+  contain Sapling outputs.
+
+- The consensus rules applied to ``valueBalance``, ``vShieldedOutput``, and ``bindingSig``
+  in non-coinbase transactions MUST also be applied to coinbase transactions.
+
+- The existing consensus rule requiring transactions that spend coinbase outputs to have
+  an empty ``vout``, is amended to only apply to transactions that spend transparent
+  coinbase outputs.
+
+- The existing consensus rule requiring coinbase outputs to have 100 confirmations before
+  they may be spent (coinbase maturity), is amended to only apply to transparent coinbase
+  outputs.
+
+- All Sapling outputs in coinbase transactions MUST have valid note commitments when
+  recovered using a 32-byte array of zeroes as the outgoing viewing key.
+
+
+Interaction with funding streams
+--------------------------------
+
+If ZIP 207 [#zip-0207]_ is accepted and deployed prior to (or in the same network upgrade
+as) this ZIP, the following additional consensus rules are enforced once the TODO network
+upgrade activates:
+
+- Funding streams MAY target Sapling addresses in addition to transparent P2SH addresses.
+
+- The existing consensus rule from ZIP 207 [#zip-0207]_ requiring coinbase transactions
+  for block height ``height`` to pay active funding streams with a standard P2SH script,
+  is amended to only apply to active funding streams that target a transparent address at
+  ``height``.
+
+- In the coinbase transaction for block height ``height``, for every active funding stream
+  that targets a Sapling address at ``height``, the transaction MUST include at least one
+  Sapling output that pays exactly the funding stream's value to the target Sapling
+  address.
+
+
+Interaction with the Founders' Reward
+-------------------------------------
+
+This ZIP does not alter the existing Founders' Reward addresses. If a future ZIP proposed
+moving to shielded Founders' Reward addresses (instead of, or before, deploying ZIP 207),
+the necessary consensus rules would be similar to those specified above for funding
+streams.
+
+
+Rationale
+=========
+
+The ZIP does not require that all coinbase must be shielded immediately from activation of
+the network upgrade, so that miners and mining pools may gradually migrate from their
+existing transparent addresses to Sapling addresses. This also simplifies the consensus
+rules, because the Founders' Reward targets transparent addresses, and thus it remains
+necessary for the time being to support them. A future ZIP could require all coinbase to
+be shielded immediately.
```

Comment:
I know you got talked down on the language on this one from Daira, but I think it's probably a good idea to indicate that we're doing this with the intent of later doing that. "So that a later ZIP can require all coinbase to be shielded immediately." perhaps.

Reply:
Seconded